stablehlo/tests/ops_stablehlo.mlir

```diff
@@ -824,6 +824,34 @@ func.func @all_to_all_c8(%data: tensor<4x16xf32>) -> tensor<16x4xf32> {
 
 // -----
 
+func.func @all_to_all_c9(%data: tensor<4x16xf32>) -> tensor<16x4xf64> {
```

Comment:
Before all the `all_to_all_cN` tests, there are some positive test cases. Let's add one for `all_to_all_variadic`, `all_gather_variadic`, and `all_reduce_variadic`.

For the all-to-all and all-gather tests, used mixed element types.

Reply:
Done. Added positive tests with variadic inputs, for the ops.